torchao/experimental/tests/test_linear_8bit_act_xbit_weight_subclass_quantizer.py

```diff
@@ -0,0 +1,153 @@
+                        nbit=nbit,
+                        has_weight_zeros=has_weight_zeros,
+                        target="fallback",
+                    ),
+                )
+
+                with torch.no_grad():
+                    result = quantized_model(activations)
+                    expected_result = quantized_model_reference(activations)
+
+                    #TODO: remove expected_result2 checks when we deprecate non-subclass API
+                    reference_impl = _Int8DynActIntxWeightQuantizedLinearFallback()
+                    reference_impl.quantize_and_pack_weights(
+                        model[0].weight, nbit, group_size, has_weight_zeros
+                    )
+                    expected_result2 = reference_impl(activations)
+
+                num_mismatch_at_low_tol = 0
+                num_mismatch_at_low_tol2 = 0
+                num_total = result.reshape(-1).shape[0]
+                for i in range(num_total):
+                    actual_val = result.reshape(-1)[i]
+                    expected_val = expected_result.reshape(-1)[i]
+                    expected_val2 = expected_result2.reshape(-1)[i]
+                    self.assertTrue(torch.allclose(actual_val, expected_val, atol=1e-6))
+                    if not torch.allclose(actual_val, expected_val):
+                        num_mismatch_at_low_tol += 1
+                    
+                    self.assertTrue(torch.allclose(expected_val, expected_val2, atol=1e-2, rtol=1e-1))
+                    if not torch.allclose(expected_val, expected_val2):
+                        num_mismatch_at_low_tol2 += 1
+
+                # Assert at most 5% of entries are not close at a low tolerance
+                self.assertTrue(num_mismatch_at_low_tol / num_total <= 0.05)
+                self.assertTrue(num_mismatch_at_low_tol2 / num_total <= 0.01)
+
+    def test_export_compile_aoti(self):
+        group_size = 32
+        m = 3
+        k0 = 512
+        k1 = 256
+        k2 = 128
+        k3 = 1024
+        nbit = 4
+        has_weight_zeros = False
+        layers = [
+            torch.nn.Linear(k0, k1, bias=False),
+            torch.nn.Linear(k1, k2, bias=False),
+            torch.nn.Linear(k2, k3, bias=False),
+        ]
+        model = torch.nn.Sequential(*layers)
+        activations = torch.randn(2, 1, m, k0, dtype=torch.float32)
+
+        print("Quantizing model")
+        quantize_(
+            model,
+            int8_dyn_act_intx_weight(
+                group_size=group_size,
+                nbit=nbit,
+                has_weight_zeros=has_weight_zeros,
+                target="native",
+            ),
+        )
+
+        unwrapped_model = copy.deepcopy(model)
+        unwrap_tensor_subclass(model)
+
+        print("Exporting quantized model")
+        exported = torch.export.export(model, (activations,))
+        
+        print("Compiling quantized model")
+        compiled = torch.compile(unwrapped_model)
```

Comment:
@jerryzh168 do you see unification for compile and export coming soon?  The fact that one requires an unwrapped tensor subclass and the other requires a wrapped one makes using this API inconvenient in torchchat.